service/controller/resource/ingress/resource.go

```diff
@@ -99,12 +99,6 @@ func toIngress(v interface{}, config Config) (metav1.Object, error) {
 		},
 		ObjectMeta: objectMeta,
 		Spec: extensionsv1beta1.IngressSpec{
-			TLS: []extensionsv1beta1.IngressTLS{
-				{
-					Hosts:      []string{config.BaseDomain},
-					SecretName: key.SecretTLSCertificates(cluster),
-				},
-			},
```

Comment:
Please leave a comment somewhere in this function/file to explain where the TLS certificate is coming from.

Reply:
Also I do not feel confident about removing this section. We have https endpoint here, so this requires TLS certificate to work. I know nginx handle this in a unexpected way and choose first certificate matching the domain, but having this here causes no harm AFAIK ?

Reply:
Well it does cause a problem in some specific cases because we create multiple ingresses each with it's own TLS config and only one can be used so it picks up a random one. But if it happens that that secret is out of date for some reason then it breaks HTTPS access. This is the problem I had on iris and some other installations over christmas and it's the whole reason for this change.

We already define the TLS certs in multiple places - PMO chart, Promxy/g8s-prometheus, here, it's a bit messy IMO and it's hard to keep track of which one takes precedence and hence which one is used.

Reply:
Please add this explanation next to the note so we remember why we did this.

Reply:
Done

Reply:
I don't see it

Reply:
Yeah this partial diff view in conversation tab isn't showing enough context. I added another comment to highlight the right bit.